File: src/main/java/upbrella/be/user/validation/OnlyNumbersValidator.java
package upbrella.be.user.validation;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

public class OnlyNumbersValidator implements ConstraintValidator<OnlyNumbers, String> {

    @Override
    public void initialize(OnlyNumbers constraintAnnotation) {
    }

    @Override
    public boolean isValid(String value, ConstraintValidatorContext context) {

        if (value == null) return true;
        return value.matches("^\\d+$");
    }
}

File: src/main/kotlin/upbrella/be/user/validation/OnlyNumbersValidator.kt
package upbrella.be.user.validation

import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class OnlyNumbersValidator : ConstraintValidator<OnlyNumbers, String> {

    override fun initialize(constraintAnnotation: OnlyNumbers) {
    }

    override fun isValid(value: String?, context: ConstraintValidatorContext): Boolean {
        if (value == null) return true
        return value.matches(Regex("^\\d+$"))
    }
}